File: app/src/models/remote.ts
/**
 * This is the magic remote name prefix
 * for when we add a remote on behalf of
 * the user.
 */
export const ForkedRemotePrefix = 'github-desktop-'

export function forkPullRequestRemoteName(remoteName: string) {
  return `${ForkedRemotePrefix}${remoteName}`
}

/** A remote as defined in Git. */
export interface IRemote {
  readonly name: string
  readonly url: string
}

/**
 * Gets a value indicating whether two remotes can be considered
 * structurally equivalent to each other.
 */
export function remoteEquals(x: IRemote | null, y: IRemote | null) {
  if (x === y) {
    return true
  }

  if (x === null || y === null) {
    return false
  }

  return x.name === y.name && x.url === y.url
}

File: GitDesktop/GitDesktop/Models/Remote.swift
import Foundation

/// Magic remote-name prefix for Desktop-created fork remotes.
/// Port of `ForkedRemotePrefix` in `models/remote.ts`.
public let forkedRemotePrefix = "github-desktop-"

/// A remote as defined in Git. Port of `IRemote`.
public struct Remote: Codable, Sendable, Equatable, Hashable, Identifiable {
    public var name: String
    public var url: String

    public init(name: String, url: String) {
        self.name = name
        self.url = url
    }

    public var id: String { name }
}

public func remoteEquals(_ x: Remote?, _ y: Remote?) -> Bool {
    switch (x, y) {
    case (nil, nil): return true
    case (nil, _), (_, nil): return false
    case (let a?, let b?): return a.name == b.name && a.url == b.url
    }
}
